### DISCUSSION TITLE
ESP32 - NMEA 0183 support

### DISCUSSION THREAD

**Discussion by big-col** [hatlabs/discussions#111] [General]
Hi, do any ESP32 hatlabs products support receiving or sending NMEA 0183?

I have a couple of older devices that do not support NMEA2K.

**Comment by mairas** [comment]
Not directly, but it's very easy to do with isolated RS485 modules such as these:

![PXL_20240401_110008305](https://github.com/hatlabs/discussions/assets/407010/ced1aadd-be95-4f8e-b9d5-dbc741608971)

You need one module for each transmission direction. The RX and TX pin of each respective module should be connected to GPIOs on the ESP32 and then you can configure ESP32 Serial1 to use those pins.

I've had a bunch of these modules lying in my (physical) inbox for a while. I can put them available on shop.hatlabs.fi in a day or two - but if you only need those modules, it's _much_ cheaper and probably faster for you to get them from Amazon, Ebay or Aliexpress. Search for "isolated rs485 module".

**Comment by big-col** [comment]
Brilliant thanks. I will order a couple from Amazon. Colin.

**Comment by mairas** [comment]
Hey, GitHub Discussions work a bit different from Issues. When they're closed, they're hidden from the default listing view. Since these questions (and their answers) are quite useful for other people to see, I'll reopen this discussion (and the other closed ones as well.)
